dataplane/vppagent/pkg/converter/remote_connection_converter.go

```diff
@@ -82,5 +82,18 @@ func (c *RemoteConnectionConverter) ToDataRequest(rv *rpc.DataRequest) (*rpc.Dat
 		},
 	})
 
+	// Process routes
+	if c.side == SOURCE {
```

Comment:
I'm a little confused here... if the c.side == SOURCE in the remoteconnection converter... then the source is remote, and we don't have the namespace here to configure the routes... what am I missing here?

Reply:
Hmps, Let me check. Based on https://github.com/ligato/networkservicemesh/blob/dc17571debd044f378408f0b83413cc973b81ac5/dataplane/vppagent/pkg/converter/cross_connect_converter.go#L40

It is really Source in case of LocalSource. And for our remote case it is so.

Reply:
Right, but its called in:

https://github.com/ligato/networkservicemesh/blob/ebdbac97d13a3ab7add93e24fc4a7accd38d5efc/dataplane/vppagent/pkg/converter/remote_connection_converter.go#L44

A NewRemoteConnectionConverter with side==SOURCE  is only called if its a RemoteSource (not a LocalSource):

https://github.com/ligato/networkservicemesh/blob/ebdbac97d13a3ab7add93e24fc4a7accd38d5efc/dataplane/vppagent/pkg/converter/cross_connect_converter.go#L50

So I'm still confused...

Reply:
Yeh, You are right, I'm also confused about this SOURCE/DEST stuff.

https://github.com/ligato/networkservicemesh/blob/ebdbac97d13a3ab7add93e24fc4a7accd38d5efc/controlplane/pkg/nsmd/network_service_server.go#L284

Here is how I've created it for Remote case, and I've put it as LocalSource, so checked this variant.

I suppose we need update Dataplane to be more clear.

Reply:
@haiodo Yeah... clarity will be good.

The basic idea is this:

SOURCE == "Who requested this connection" ie, an NSC
DESTINATION == "Who was connected to in response to the request" ie, an NSE.

ConnectionContext gets set in the SOURCE, but only if the SOURCE is Local (if its remote, we can't meaningfully set it, its just a tunnel) and *only* if the Mechanism type of the LocalSource is KernelInterface (if its Memif, its not our responsibility).

Reply:
Checked, in remote case indeed this is not required. Thanks.